src/types/form_logic.ts

```diff
@@ -107,6 +125,61 @@ export type LogicCondition =
   | CategoricalLogicCondition
   | BinaryLogicCondition
   | NumericalLogicCondition
+  | MultiValuedLogicCondition
+
+/**
+ * Types needed for logic module inputs
+ */
+
+// Type of client logic fields before transformation (passed into logic module).
+// We don't store a fieldValue in the database, but the client
+// needs it as a variable to store the client's answer to a field.
+export interface ILogicInputClientSchema extends IFieldSchema {
+  fieldValue: string | boolean[]
+}
+
+// Type for fields that are passed into the logic module
+export type FieldSchemaOrResponse = ILogicInputClientSchema | FieldResponse
+
+// Type for client logic fields after transformation
+export interface ILogicClientFieldSchema
+  extends Omit<ILogicInputClientSchema, 'fieldValue'> {
+  // Use omit instead of directly extending IFieldSchema
+  // to prevent typescript from complaining about return type in adaptor function
+  fieldValue: string | CheckboxConditionValue
```

Comment:
I don’t think this will solve the problem because we will still have to assign the value to the field object in the end, which will ultimately require the definition of a type for it.